rsocket-transport-netty/src/main/java/io/rsocket/transport/netty/WebsocketUriHandler.java

```diff
@@ -22,26 +22,39 @@
 import io.rsocket.transport.netty.server.WebsocketServerTransport;
 import io.rsocket.uri.UriHandler;
 import java.net.URI;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
 import java.util.Optional;
 
-public class WebsocketUriHandler implements UriHandler {
+/**
+ * An implementation of {@link UriHandler} that creates {@link WebsocketClientTransport}s and {@link
+ * WebsocketServerTransport}s.
+ */
+public final class WebsocketUriHandler implements UriHandler {
+
+  private static final List<String> SCHEME = Arrays.asList("ws", "wss");
+
   @Override
   public Optional<ClientTransport> buildClient(URI uri) {
-    if ("ws".equals(uri.getScheme()) || "wss".equals(uri.getScheme())) {
-      return Optional.of(WebsocketClientTransport.create(uri));
+    Objects.requireNonNull(uri, "uri must not be null");
+
+    if (SCHEME.stream().noneMatch(scheme -> scheme.equals(uri.getScheme()))) {
+      return Optional.empty();
     }
 
-    return UriHandler.super.buildClient(uri);
+    return Optional.of(WebsocketClientTransport.create(uri));
   }
 
   @Override
   public Optional<ServerTransport> buildServer(URI uri) {
-    if ("ws".equals(uri.getScheme())) {
-      return Optional.of(
-          WebsocketServerTransport.create(
-              uri.getHost(), WebsocketClientTransport.getPort(uri, 80)));
+    Objects.requireNonNull(uri, "uri must not be null");
+
+    if (SCHEME.stream().noneMatch(scheme -> scheme.equals(uri.getScheme()))) {
+      return Optional.empty();
     }
 
-    return UriHandler.super.buildServer(uri);
+    return Optional.of(
+        WebsocketServerTransport.create(uri.getHost(), WebsocketClientTransport.getPort(uri, 80)));
```

Comment:
I believe this will cause a wss://rsocket.mycompany.com/endpoint to use 80 instead of 443

Reply:
That’s [always been](https://github.com/rsocket/rsocket-java/pull/489/files/acb25677a0af0b508e3a575aa9d1a18c77a4c35c#diff-f703148c739a07205efcd466989696d5L42) the case though. In this update, I’m aiming to not change any of the behavior as it exists.  If we think it’s a bug, maybe opening an issue that I’ll work separately?

Reply:
Before it was only supporting plaintext as a server, so 80 was correct.  IIRC my assumption at the time was it's easy to create a client with default SslContext etc, but servers generally require additional setup e.g. cert chains.